### DISCUSSION TITLE
[Feature] Machine Learning using Intel NPU in OpenVINO on supported Core Ultra Processors

### DISCUSSION THREAD

**Discussion by volschin** [immich-app/immich#14583] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

I have installed the drivers regarding the Release notes on my Ubuntu 24.04.
- https://github.com/intel/linux-npu-driver/releases/tag/v1.10.0

The /dev/accel/accel0 device is available on the host system now. I attached it to the Immich machine learning image by adding a further device with
- /dev/accel:/dev/accel

This doesn‘t seem to be enough, to enable it in the Immich ml image for openvino.
There will be more and more of the Core Ultra CPU with integrated NPU on the market.

it would be nice to have this working in the container.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by lory9995** [comment]
I’d love this too

**Comment by absamue** [comment]
+1

As a comparison, using the NPU for object detection is recommended by Scrypted NVR: https://docs.scrypted.app/buyers-guide/servers.html#nvr-recommendations

**Comment by KingCZE** [comment]
Another +1 here. 
It would be nice to use the low-power NPU instead of a power-hungry GPU, CPU or not very performant iGPU.

**Comment by janchanru** [comment]
Good afternoon! Any chances of adding AMD NPU support? I have an AMD Ryzen 7 7640HS with NPU, would love to see it utilized in Immich. It could drastically reduce power consumption compared to GPU.

**Comment by sneak-it** [comment]
This would be a really cool addition. Lots more CPU's that have built in NPU's coming into the (homelab) market, this would be a welcome addition!

**Comment by Tubare** [comment]
I just installed a NAS that uses an ultra series CPU, and I also hope to use the npu！

**Comment by duckdude555** [comment]
+1, NPU support would be awesome!

**Comment by bo0tzz** [comment]
Folks, if you have no substance to add please just upvote the original post.

**Comment by EncryptedCurse** [comment]
Please correct me if I'm wrong, but it looks trivial to add NPU support:

https://github.com/immich-app/immich/blob/4ffa26c9695a5bdbf4d5b4d7d49841955efe9408/machine-learning/immich_ml/sessions/ort.py#L96-L110

With that said, it would certainly be useful to have an environment variable to control which device gets used.

I'll try to put together a PR if I find the time.
